plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -200,59 +195,49 @@ public String toString() {
 		return IAuthorizationContext.AUTHSCHEME_OAUTH;
 	}
 	
-	private class AuthLinkHandler extends SelectionAdapter{
-		private String link = "";
-		private Shell shell;
-
-		AuthLinkHandler(Shell shell, IAuthorizationDetails details){
-			this.shell = shell;
-			if(details != null) {
-				details.getRequestTokenLink();
-			}
-		}
+	private Listener onRetrieveLinkClicked(final Shell shell) {
+		return new Listener() {
 
-		@Override
-		public void widgetSelected(SelectionEvent e) {
-			if(!StringUtils.isBlank(link)) {
```

Comment:
Trying to recall, but I believe this is the chicken/egg problem (or was).  Ideally we want to present the link to the user with the actual URL, but we don't know what it is until we try to authenticate and error.  The error returns the link to where to go and get the token.  I'm not sure why we don't do anything with the return value of 'details.getRequestTokenLink();' in the previous block other then to trigger the auth cycle to actually GET the link we need; it otherwise is not useful and could be an artifact that never was removed.